### ISSUE TITLE
iterating though exif keys with im.get() raises NotImplementedError

### ISSUE THREAD

**Issue by pjgoodall** [TNThieding/exif#30]
I cannot iterate through the 'complete' values and keys of an image either with a for loop or a list comprehension. Can someone check please.

``` bash
% python --version
>Python 3.7.6
```

The usage guide says a non-existent key should return `None`:

``` text
>>> my_image.get("color_space")
<ColorSpace.UNCALIBRATED: 65535>
>>> my_image.get("nonexistent_tag")
None
```

## Source 1
``` python
from exif import Image
from pathlib import Path

# https://upload.wikimedia.org/wikipedia/commons/5/5d/Pipe_Creek_Vista-Grand_Canyon.jpg
pic_path = Path("Pipe_Creek_Vista-Grand_Canyon.jpg")

answer = {}
with pic_path.open("rb") as f:
    im = Image(f)
    keys = dir(im)
    for k in keys:
        print(f"key = {k}")
        print(f"value = {im.get(k)}")
```
# Source 2
``` python
from exif import Image
from pathlib import Path

# https://upload.wikimedia.org/wikipedia/commons/5/5d/Pipe_Creek_Vista-Grand_Canyon.jpg
pic_path = Path("Pipe_Creek_Vista-Grand_Canyon.jpg")
with pic_path.open("rb") as f:
    im = Image(f)
    keys = dir(im)
    list = [im.get(key) for key in keys]
    table = dict(zip(keys, list))
print(table)
```
## Output of source 1
``` shell
key = _exif_ifd_pointer
value = 170
key = _segments
value = None
key = color_space
value = 1
key = components_configuration
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-5-3c91ddaf2d67> in <module>
     11     for k in keys:
     12         print(f"key = {k}")
---> 13         print(f"value = {im.get(k)}")

~/miniconda3/envs/jpeg_exif/lib/python3.7/site-packages/exif/_image.py in get(self, attribute, default)
    124         """
    125         try:
--> 126             retval = self.__getattr__(attribute)
    127         except AttributeError:
    128             retval = default

~/miniconda3/envs/jpeg_exif/lib/python3.7/site-packages/exif/_image.py in __getattr__(self, item)
     76 
     77     def __getattr__(self, item):
---> 78         return getattr(self._segments['APP1'], item)
     79 
     80     def __setattr__(self, key, value):

~/miniconda3/envs/jpeg_exif/lib/python3.7/site-packages/exif/_app1_metadata.py in __getattr__(self, item)
    256             raise AttributeError(ERROR_IMG_NO_ATTR.format(item))
    257 
--> 258         return ifd_tag.read()
    259 
    260     def __setattr__(self, key, value):

~/miniconda3/envs/jpeg_exif/lib/python3.7/site-packages/exif/_ifd_tag/_base.py in read(self)
     63 
     64         """
---> 65         raise NotImplementedError("cannot read a base/unknown IFD tag instance")

NotImplementedError: cannot read a base/unknown IFD tag instance
```

[comprehension_error.zip](https://github.com/TNThieding/exif/files/4700460/comprehension_error.zip)

**Comment by TNThieding** [comment]
Thank you for bringing this to my attention. I believe I fixed this issue in ``v0.8.6`` of the package, which I released minutes ago. Now, ``get()`` returns the default value  if the specified tag cannot be read. If you still run into issues, feel free to open a new issue. Thank you!

**Comment by pjgoodall** [comment]
Thanks Tyler
